### ISSUE TITLE
JAXB with JDK > 8

### ISSUE THREAD

**Issue by aharth** [kaefer3000/ldbbc#2]
Looks like LDBBC throws exceptions when using JDK > 8, because JAXB is not shipped any more since Java 9.

Looks like the Maven script has to be adapted, e.g., https://jesperdj.com/2018/09/30/jaxb-on-java-9-10-11-and-beyond/.

/cc @sebastian-lempert

**Comment by kaefer3000** [comment]
I am curious to see your exception as I wonder which part requires JAXB (maybe the RDF/XML code from NxParser?).

**Comment by sebastian-lempert** [comment]
Please find attached and extract from the exceptions that did run into: [LDBBC exceptions.txt](https://github.com/kaefer3000/ldbbc/files/5607229/LDBBC.exceptions.txt)

I was testing on macOS 10.15.7 Catalina with Java 15. On my machine

```bash
¢ java --version
```

gives:

```
java version "15" 2020-09-15
Java(TM) SE Runtime Environment (build 15+36-1562)
Java HotSpot(TM) 64-Bit Server VM (build 15+36-1562, mixed mode, sharing)
```

**Comment by kaefer3000** [comment]
Thanks. So after one POST request, exceptions keep coming in infinitely? Can you provide some details about the POST request (eg. a CURL command to reproduce)?

**Comment by sebastian-lempert** [comment]
Exactly. Unfortunately I did these tests some time ago and I forgot to document the corresponding commands being used. If I remember correctly I was trying to run the LDFU based coinflip example [1] that @aharth provided to me in combination with your [coinflip-servlet](https://github.com/kaefer3000/coinflip-servlet). I just followed the corresponding README.md. Thus please see this README.md for the shell commands being used and for the requests being executed see the corresponding N3 programs that were executed with LDFU. 

[1] LDFU based coinflip example: [coinflip.zip](https://github.com/kaefer3000/ldbbc/files/5607344/coinflip.zip)

**Comment by sebastian-lempert** [comment]
Btw: the coinflip example from above runs fine with LDBBC and Jetty on a Raspberry Pi 4 running Raspbian GNU/Linux 10 (buster) and Java 8 where

```bash
java -version
```

gives

```
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (build 1.8.0_212-8u212-b01-1+rpi1-b01)
OpenJDK Client VM (build 25.212-b01, mixed mode)
```

**Comment by kaefer3000** [comment]
This required quite some refactoring here and in [NxParser](https://github.com/nxparser/nxparser.git). Jaxb is not the issue, this is sorted with Java 11-compatible JavaEE dependencies. But those new JavaEE dependencies require new namespaces, as [JavaEE has been transferred from Oracle to Eclipse](https://blogs.oracle.com/javamagazine/transition-from-java-ee-to-jakarta-ee) and Oracle has kept the packages' namespace. Check out https://github.com/kaefer3000/ldbbc/releases/tag/0.0.7

**Comment by sebastian-lempert** [comment]
Thanks for your help. From your description it is also clear to me why simply changing the pom.xml as stated at https://jesperdj.com/2018/09/30/jaxb-on-java-9-10-11-and-beyond/ was not sufficient. I had found this page when I was looking for solutions to the original exceptions. Therefore only the assumption that it could have something to do with JAXB.

I could successfully test the new version 0.0.7 of LDBBC. However, I had to use the current ZIP archive [0.0.7.zip](https://github.com/kaefer3000/ldbbc/archive/0.0.7.zip) of the source code.

When I check out the source code via

```bash
$ git clone https://github.com/kaefer3000/coinflip-servlet.git
```

it seems that source code for version 0.0.7 is not yet delivered, correct?

What I noticed while running the coinflip example with your new LDBBC version 0.0.7 from the ZIP archive mentioned above:
1. The LDBBC seems to work as expected, but produces a very long list of warnings right after startup: 
[LDBBC - warnings.txt](https://github.com/kaefer3000/ldbbc/files/5610670/LDBBC.-.warnings.txt)
1. The coinflip-servlet can be started normally, but once produces a long list of exceptions when I run ```watch -n 1 roqet heads-roqet.rq```. Afterwards the coinflip-servlet seems to function normally. Here are the corresponding exceptions: 
[coinflip-servlet - exceptions.txt](https://github.com/kaefer3000/ldbbc/files/5610694/coinflip-servlet.-.exceptions.txt)

**Comment by kaefer3000** [comment]
The problem was indeed partially with Jaxb, the question is what is the best place to fix the problem.

The source code is in a branch, https://github.com/kaefer3000/ldbbc/tree/java11-jakarta .

1. I hope they go away once Jetty etc. are stable
2. The coinflip-servlet is based on the same code for the REST part as ldbbc. Hence, the same java11 exceptions as with ldbbc, seems like they are not fatal for GET requests. The update is a piece of cake (version bump nxparser and jetty in pom.xml; repacing `javax.*` with `jakarta.*` in the classes).

**Comment by kaefer3000** [comment]
For the coinflip servlet and java11 / jakarta, see https://github.com/kaefer3000/coinflip-servlet/tree/java11-jakarta
